CorsixTH/Src/th_lua.cpp

```diff
@@ -320,58 +320,47 @@ void luaT_pushtablebool(lua_State* L, const char* k, bool v) {
   lua_settable(L, -3);
 }
 
+void luaT_printvalue(lua_State* L, int idx) {
+  int t = lua_type(L, idx);
+  switch (t) {
+    case LUA_TSTRING: /* strings */
+      std::printf("string: '%s'\n", lua_tostring(L, idx));
+      break;
+    case LUA_TBOOLEAN: /* booleans */
+      std::printf("boolean %s\n", lua_toboolean(L, idx) ? "true" : "false");
+      break;
+    case LUA_TNUMBER: /* numbers */
+      std::printf("number: %g\n", lua_tonumber(L, idx));
+      break;
+    default: /* other values */
+      std::printf("%s: %p\n", lua_typename(L, t), lua_topointer(L, idx));
+      break;
+  }
+}
+
 void luaT_printstack(lua_State* L) {
   int i;
   int top = lua_gettop(L);
 
   std::printf("total items in stack %d\n", top);
 
   for (i = 1; i <= top; i++) { /* repeat for each level */
-    int t = lua_type(L, i);
-    switch (t) {
-      case LUA_TSTRING: /* strings */
-        std::printf("string: '%s'\n", lua_tostring(L, i));
-        break;
-      case LUA_TBOOLEAN: /* booleans */
-        std::printf("boolean %s\n", lua_toboolean(L, i) ? "true" : "false");
-        break;
-      case LUA_TNUMBER: /* numbers */
-        std::printf("number: %g\n", lua_tonumber(L, i));
-        break;
-      default: /* other values */
-        std::printf("%s\n", lua_typename(L, t));
-        break;
-    }
-    std::printf(" "); /* put a separator */
+    printf("(%d) ", i);
```

Comment:
Something debugging?

Reply:
The whole function is for debugging, along with the one above it. Otherwise looking at the lua state in gdb is hard.

Reply:
Should switch to std::printf